src/Aspire.Hosting/ApplicationModel/EndpointReference.cs

```diff
@@ -81,6 +82,11 @@ public EndpointReferenceExpression Property(EndpointProperty property)
     /// </summary>
     public int? TargetPort => EndpointAnnotation.TargetPort;
 
+    /// <summary>
+    /// Returns the port to use when referencing this endpoint from a container.
+    /// </summary>
+    public int? ContainerPort => Resource.IsContainer() ? EndpointAnnotation.TargetPort : AllocatedEndpoint.Port;
```

Comment:
`ContainerClientPort` maybe? (I am really struggling to find a good name for this)

There is a big assumption behind this implementation, namely, that all containers are running on a single network.  In future we will want to relax this, but this is probably the best we can do for Aspire 9. We should discuss the deprecation / plan for Endpoint.v2 API before we finalize this.

Reply:
I'm torn; I thought about a more explicit name that implies this is specifically a port to be used to access the endpoint from within a container, but we have the existing `ContainerHost` property that effectively serves the same purpose, so I erred on the side of keeping the new property name consistent.

I'll admit I have a suspicion that we'll end up wanting to always have a default common shared network for our containers, with any future support for custom networks being additive, but I think we should be able to get a better understanding of the full pros and cons and where we want to take container based service discovery in the long term with some discussion in the next few days